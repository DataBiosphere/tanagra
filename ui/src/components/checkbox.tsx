import CheckBoxIcon from "@mui/icons-material/CheckBox";
import CheckBoxOutlineBlankIcon from "@mui/icons-material/CheckBoxOutlineBlank";
import IconButton from "@mui/material/IconButton";

export type CheckboxProps = {
  checked?: boolean;
  onChange?: () => void;
  size?: "small" | "medium" | "large";
  fontSize?: "small" | "medium" | "large" | "inherit";
  name?: string;
};

export default function Checkbox({
  checked,
  onChange,
  size,
  fontSize,
  name,
}: CheckboxProps) {
  return (
    <IconButton
      role={"checkbox"}
      size={size}
      name={name}
      onClick={() => {
        if (onChange) {
          onChange();
        }
      }}
    >
      {checked ? (
        <CheckBoxIcon fontSize={fontSize} color="primary" />
      ) : (
        <CheckBoxOutlineBlankIcon
          fontSize={fontSize}
          sx={{ fill: "inherit" }}
        />
      )}
    </IconButton>
  );
}
