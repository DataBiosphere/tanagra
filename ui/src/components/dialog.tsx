import Button from "@mui/material/Button";
import Dialog from "@mui/material/Dialog";
import DialogActions from "@mui/material/DialogActions";
import DialogContent from "@mui/material/DialogContent";
import DialogTitle from "@mui/material/DialogTitle";
import TextField from "@mui/material/TextField";
import { ChangeEvent, ReactNode, useState } from "react";

type NewDialogProps = {
  title: string;
  titleId: string;
  textLabel: string;
  className: string;
  callback: (name: string) => void;
};

export function useDialog(props: NewDialogProps): [ReactNode, () => void] {
  const [open, setOpen] = useState(false);
  const show = () => {
    setOpen(true);
  };

  const [name, setName] = useState(props.title);
  const onNameChange = (event: ChangeEvent<HTMLInputElement>) => {
    setName(event.target.value);
  };

  const onCreate = () => {
    setOpen(false);
    props.callback(name);
  };

  return [
    // eslint-disable-next-line react/jsx-key
    <Dialog
      open={open}
      onClose={() => {
        setOpen(false);
      }}
      aria-labelledby={props.titleId}
      maxWidth="sm"
      fullWidth
      className={props.className}
    >
      <DialogTitle id={props.titleId}>{props.title}</DialogTitle>
      <DialogContent>
        <TextField
          autoFocus
          margin="dense"
          id="name"
          label={props.textLabel}
          fullWidth
          variant="standard"
          value={name}
          onChange={onNameChange}
        />
      </DialogContent>
      <DialogActions>
        <Button
          variant="contained"
          disabled={name.length === 0}
          onClick={onCreate}
        >
          Create
        </Button>
      </DialogActions>
    </Dialog>,
    show,
  ];
}
