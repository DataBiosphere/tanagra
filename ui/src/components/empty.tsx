import Stack from "@mui/material/Stack";
import Typography from "@mui/material/Typography";
import React from "react";

type EmptyProps = {
  title?: string;
  subtitle?: string;
  image?: string;
  maxWidth?: string | number;
  minHeight?: string | number;
};

export default function Empty(props: EmptyProps) {
  return (
    <Stack
      alignItems="center"
      justifyContent="center"
      sx={{ width: "100%", height: "100%", minHeight: props.minHeight }}
    >
      <Stack
        alignItems="center"
        sx={{ maxWidth: props.maxWidth, textAlign: "center" }}
      >
        {props.image && (
          <img src={props.image} style={{ marginBottom: "20px" }} />
        )}
        {props.title && (
          <Typography variant="body1em">{props.title}</Typography>
        )}
        {props.subtitle && (
          <Typography variant="body1">{props.subtitle}</Typography>
        )}
      </Stack>
    </Stack>
  );
}
