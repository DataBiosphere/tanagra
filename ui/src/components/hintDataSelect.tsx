import Box from "@mui/material/Box";
import Chip from "@mui/material/Chip";
import FormControl from "@mui/material/FormControl";
import MenuItem from "@mui/material/MenuItem";
import OutlinedInput from "@mui/material/OutlinedInput";
import Select, { SelectChangeEvent } from "@mui/material/Select";
import Typography from "@mui/material/Typography";
import { EnumHintOption, HintData } from "data/source";
import { DataValue } from "data/types";
import React from "react";
import { isValid } from "util/valid";

export type Selection = {
  value: DataValue;
  name: string;
};

export type HintDataSelectProps = {
  hintData?: HintData;
  selected?: Selection[];
  maxChips?: number;
  onSelect?: (sel: Selection[]) => void;
};

export function HintDataSelect(props: HintDataSelectProps) {
  const update = (sel: string[]) => {
    props.onSelect?.(
      sel
        .map((name) => {
          let value = props.hintData?.enumHintOptions?.find(
            (hint: EnumHintOption) => hint.name === name
          )?.value;
          if (value === undefined) {
            return undefined;
          }
          return {
            name,
            value: typeof value === "bigint" ? value.toString() : value,
          };
        })
        .filter(isValid)
    );
  };

  const onSelect = (event: SelectChangeEvent<string[]>) => {
    const {
      target: { value: sel },
    } = event;
    if (typeof sel === "string") {
      // This case is only for selects with text input.
      return;
    }
    update(sel);
  };

  const onDelete = (name: string) => {
    update(props.selected?.map((s) => s.name)?.filter((n) => n !== name) ?? []);
  };

  return (
    <FormControl variant="outlined">
      <Select
        variant="outlined"
        multiple
        displayEmpty
        disabled={!props.hintData}
        value={props.selected?.map((s) => s.name)}
        input={<OutlinedInput />}
        renderValue={(selected) => (
          <Box sx={{ mt: 0.25, display: "flex", flexWrap: "wrap", gap: 0.5 }}>
            {selected?.length ? (
              !props.maxChips || selected.length <= props.maxChips ? (
                selected.map((s) => (
                  <Chip
                    key={s}
                    label={s}
                    onDelete={() => {
                      onDelete(s);
                    }}
                  />
                ))
              ) : (
                <Typography variant="overline" component="em">
                  {selected.length} selected
                </Typography>
              )
            ) : (
              <Typography variant="overline" component="em">
                Any value
              </Typography>
            )}
          </Box>
        )}
        onChange={onSelect}
      >
        {props.hintData?.enumHintOptions?.map((hint: EnumHintOption) => (
          <MenuItem key={hint.name} value={hint.name}>
            {hint.name}
          </MenuItem>
        ))}
      </Select>
    </FormControl>
  );
}
