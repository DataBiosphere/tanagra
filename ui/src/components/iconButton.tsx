import { Palette, Theme } from "@mui/material/styles";

// Only allow the palette entries that are regular color palettes.
export type ColorsFromPalette = Pick<Palette, "primary" | "warning" | "error">;

// The current MUI version doesn't support "contained" IconButtons like Button
// does so add reusable styling to emulate it.
export function containedIconButtonSx(palette: keyof ColorsFromPalette) {
  return {
    "&.MuiIconButton-root": {
      backgroundColor: (theme: Theme) => theme.palette[palette].main,
      color: (theme: Theme) => theme.palette[palette].contrastText,
    },
    "&.MuiIconButton-root:hover": {
      backgroundColor: (theme: Theme) => theme.palette[palette].dark,
    },
  };
}
