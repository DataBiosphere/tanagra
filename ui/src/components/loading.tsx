import ErrorIcon from "@mui/icons-material/Error";
import Box from "@mui/material/Box";
import Button from "@mui/material/Button";
import CircularProgress from "@mui/material/CircularProgress";
import Tooltip from "@mui/material/Tooltip";
import Typography from "@mui/material/Typography";
import React, { ReactNode, useEffect, useRef, useState } from "react";

type Status = {
  error?: Error;
  isPending?: boolean;
  reload?: () => void;
};

type Props = {
  size?: string;
  status?: Status;
  children?: React.ReactNode;
};

export default function Loading(props: Props) {
  const [visible, setVisible] = useState(false);
  const timerRef = useRef<number>();

  useEffect(() => {
    if (!props.status?.isPending) {
      return;
    }

    timerRef.current = window.setTimeout(() => {
      setVisible(true);
    }, 800);

    return () => {
      setVisible(false);
      clearTimeout(timerRef.current);
    };
  }, [props.status?.isPending]);

  if (props.status && !props.status.isPending && !props.status.error) {
    return <>{props.children}</>;
  }

  return (
    <Box className={props.size === "small" ? "" : "loading"}>
      {showStatus(visible, props.status, props.size)}
    </Box>
  );
}

function showStatus(
  visible: boolean,
  status?: Status,
  size?: string
): ReactNode {
  if (status?.error && !status?.isPending) {
    return size === "small" ? (
      <Tooltip title={status.error?.message} arrow={true}>
        <ErrorIcon />
      </Tooltip>
    ) : (
      <>
        <Typography variant="h2">Error</Typography>
        {status.error?.message ? (
          <Typography paragraph>
            An error has occurred: <b>{status.error.message}</b>
          </Typography>
        ) : (
          <Typography paragraph>An unknown error has occurred.</Typography>
        )}
        <div>
          <Button onClick={status?.reload} variant="contained">
            Reload
          </Button>
        </div>
      </>
    );
  }
  return visible ? (
    <CircularProgress
      style={size === "small" ? { width: "1em", height: "1em" } : {}}
    />
  ) : null;
}
