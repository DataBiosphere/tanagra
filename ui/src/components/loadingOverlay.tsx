import CircularProgress from "@mui/material/CircularProgress";
import Modal from "@mui/material/Modal";
import Stack from "@mui/material/Stack";

export default function LoadingOverlay() {
  return (
    <Modal open sx={{ zIndex: (theme) => theme.zIndex.appBar - 1 }}>
      <Stack
        alignItems="center"
        justifyContent="center"
        sx={{
          width: "100%",
          height: "100%",
        }}
      >
        <CircularProgress />
      </Stack>
    </Modal>
  );
}
