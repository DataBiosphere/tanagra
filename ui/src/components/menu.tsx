import {
  default as BaseMenu,
  MenuProps as BaseMenuProps,
} from "@mui/material/Menu";
import { MouseEvent, ReactElement, useState } from "react";

type MenuProps = Omit<
  BaseMenuProps,
  "anchorEl" | "open" | "onClick" | "onClose"
>;

export function useMenu({
  children,
  ...props
}: MenuProps,
clickNclose = true): [
  ReactElement,
  (e: MouseEvent<HTMLElement | undefined>) => void
] {
  const [anchorEl, setAnchorEl] = useState<HTMLElement | undefined>();
  const show = (e: MouseEvent<HTMLElement | undefined>) =>
    setAnchorEl(e.currentTarget);

  return [
    // eslint-disable-next-line react/jsx-key
    <BaseMenu
      {...props}
      anchorEl={anchorEl}
      open={!!anchorEl}
      onClick={() => {if (clickNclose) {setAnchorEl(undefined)}}}
      onClose={() => setAnchorEl(undefined)}
    >
      {children}
    </BaseMenu>,
    show,
  ];
}
