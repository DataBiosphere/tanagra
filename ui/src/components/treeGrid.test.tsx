import AccountTreeIcon from "@mui/icons-material/AccountTree";
import CheckBoxIcon from "@mui/icons-material/CheckBox";
import IconButton from "@mui/material/IconButton";
import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import { TreeGrid, TreeGridData, TreeGridId } from "components/treeGrid";

test("Table renders correctly", async () => {
  const columns = [
    {
      key: "col1",
      width: "100%",
      title: "Column 1",
      sortable: true,
    },
    {
      key: "col2",
      width: 150,
      title: "Column 2",
      sortable: true,
    },
    {
      key: "col3",
      width: 50,
      title: <AccountTreeIcon />,
    },
  ];

  const data: TreeGridData = {
    children: [1, 3],
    rows: new Map([
      [
        1,
        {
          data: {
            col1: "1-col1",
            col2: "1-col2",
          },
          children: [2],
        },
      ],
      [
        2,
        {
          data: {
            col1: "2-col1",
            col2: "2-col2",
            col3: "2-col3",
          },
        },
      ],
      [
        3,
        {
          data: {
            col1: "3-col1",
            col2: "3-col2",
            col3: <AccountTreeIcon />,
          },
        },
      ],
    ]),
  };

  const rowCustomization = (id: TreeGridId) => {
    // Add variety to the rows.
    if (id !== 2) {
      return undefined;
    }
    return [
      {
        column: 0,
        prefixElements: (
          <IconButton size="small">
            <CheckBoxIcon fontSize="inherit" />
          </IconButton>
        ),
      },
    ];
  };

  const onSort = vitest.fn();
  const { asFragment } = render(
    <TreeGrid
      columns={columns}
      data={data}
      sortOrders={[]}
      onSort={onSort}
      rowCustomization={rowCustomization}
    />
  );
  expect(asFragment()).toMatchSnapshot();

  const sortIcons = await screen.findAllByTestId("SwapVertIcon");
  expect(sortIcons.length).toBe(2);

  await userEvent.click(sortIcons[0]);
  expect(asFragment()).toMatchSnapshot();
  expect(onSort).toHaveBeenCalled();
});
