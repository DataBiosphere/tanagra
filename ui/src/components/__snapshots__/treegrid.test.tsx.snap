// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Table renders correctly 1`] = `
<DocumentFragment>
  <div
    class="css-1qqflhf"
  >
    <table
      style="table-layout: fixed; width: 100%; text-align: left; border-collapse: collapse;"
    >
      <thead>
        <tr
          style="height: 48px;"
        >
          <th
            style="position: sticky; top: 0px; max-width: 100%; width: 100%; min-width: 100%; background-color: rgb(255, 255, 255); box-shadow: inset 0 -2px 0 rgba(0, 0, 0, 0.12); z-index: 1;"
          >
            <div
              class="MuiBox-root css-semc09"
            >
              <div
                class="MuiBox-root css-j6zvzo"
              >
                <span
                  class="MuiTypography-root MuiTypography-overline css-szitil-MuiTypography-root"
                  title="Column 1"
                >
                  Column 1
                </span>
              </div>
              <div
                class="MuiBox-root css-9qc0eg"
              >
                <button
                  class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeSmall css-1pe4mpk-MuiButtonBase-root-MuiIconButton-root"
                  tabindex="0"
                  type="button"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1vooibu-MuiSvgIcon-root"
                    data-testid="SwapVertIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M16 17.01V10h-2v7.01h-3L15 21l4-3.99h-3zM9 3 5 6.99h3V14h2V6.99h3L9 3z"
                    />
                  </svg>
                  <span
                    class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                  />
                </button>
              </div>
            </div>
          </th>
          <th
            style="position: sticky; top: 0px; max-width: 150px; width: 150px; min-width: 150px; background-color: rgb(255, 255, 255); box-shadow: inset 0 -2px 0 rgba(0, 0, 0, 0.12); z-index: 1;"
          >
            <div
              class="MuiBox-root css-semc09"
            >
              <div
                class="MuiBox-root css-j6zvzo"
              >
                <span
                  class="MuiTypography-root MuiTypography-overline css-szitil-MuiTypography-root"
                  title="Column 2"
                >
                  Column 2
                </span>
              </div>
              <div
                class="MuiBox-root css-9qc0eg"
              >
                <button
                  class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeSmall css-1pe4mpk-MuiButtonBase-root-MuiIconButton-root"
                  tabindex="0"
                  type="button"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1vooibu-MuiSvgIcon-root"
                    data-testid="SwapVertIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M16 17.01V10h-2v7.01h-3L15 21l4-3.99h-3zM9 3 5 6.99h3V14h2V6.99h3L9 3z"
                    />
                  </svg>
                  <span
                    class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                  />
                </button>
              </div>
            </div>
          </th>
          <th
            style="position: sticky; top: 0px; max-width: 50px; width: 50px; min-width: 50px; background-color: rgb(255, 255, 255); box-shadow: inset 0 -2px 0 rgba(0, 0, 0, 0.12); z-index: 1;"
          >
            <div
              class="MuiBox-root css-1jjjjm8"
            >
              <div
                class="MuiBox-root css-j6zvzo"
              >
                <span
                  class="MuiTypography-root MuiTypography-overline css-szitil-MuiTypography-root"
                  title="[object Object]"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                    data-testid="AccountTreeIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M22 11V3h-7v3H9V3H2v8h7V8h2v10h4v3h7v-8h-7v3h-2V8h2v3z"
                    />
                  </svg>
                </span>
              </div>
            </div>
          </th>
        </tr>
      </thead>
      <tbody>
        <tr
          style="height: 48px;"
        >
          <td
            style="max-width: 100%; width: 100%; min-width: 100%;"
          >
            <div
              class="MuiStack-root css-1sg8hwy-MuiStack-root"
            >
              <button
                class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeSmall css-1pe4mpk-MuiButtonBase-root-MuiIconButton-root"
                tabindex="0"
                type="button"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1vooibu-MuiSvgIcon-root"
                  data-testid="KeyboardArrowRightIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M8.59 16.59 13.17 12 8.59 7.41 10 6l6 6-6 6-1.41-1.41z"
                  />
                </svg>
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </button>
              <button
                class="MuiTypography-root MuiTypography-body2 MuiLink-root MuiLink-underlineNone MuiLink-button css-14wmvzk-MuiTypography-root-MuiLink-root"
                title="1-col1"
              >
                1-col1
              </button>
            </div>
          </td>
          <td
            style="max-width: 150px; width: 150px; min-width: 150px;"
          >
            <div
              class="MuiStack-root css-1d9cypr-MuiStack-root"
            >
              <p
                class="MuiTypography-root MuiTypography-body2 css-foitbu-MuiTypography-root"
                title="1-col2"
              >
                1-col2
              </p>
            </div>
          </td>
          <td
            style="max-width: 50px; width: 50px; min-width: 50px;"
          >
            <div
              class="MuiStack-root css-1d9cypr-MuiStack-root"
            >
              <p
                class="MuiTypography-root MuiTypography-body2 css-foitbu-MuiTypography-root"
                title=""
              />
            </div>
          </td>
        </tr>
        <tr
          style="display: none; height: 48px;"
        >
          <td
            style="max-width: 100%; width: 100%; min-width: 100%; box-shadow: inset 0 1px 0 rgba(0, 0, 0, 0.12);"
          >
            <div
              class="MuiStack-root css-kv8qlv-MuiStack-root"
            >
              <button
                class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeSmall css-1pe4mpk-MuiButtonBase-root-MuiIconButton-root"
                tabindex="0"
                type="button"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1vooibu-MuiSvgIcon-root"
                  data-testid="CheckBoxIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M19 3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.11 0 2-.9 2-2V5c0-1.1-.89-2-2-2zm-9 14-5-5 1.41-1.41L10 14.17l7.59-7.59L19 8l-9 9z"
                  />
                </svg>
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </button>
              <p
                class="MuiTypography-root MuiTypography-body2 css-foitbu-MuiTypography-root"
                title="2-col1"
              >
                2-col1
              </p>
            </div>
          </td>
          <td
            style="max-width: 150px; width: 150px; min-width: 150px; box-shadow: inset 0 1px 0 rgba(0, 0, 0, 0.12);"
          >
            <div
              class="MuiStack-root css-1d9cypr-MuiStack-root"
            >
              <p
                class="MuiTypography-root MuiTypography-body2 css-foitbu-MuiTypography-root"
                title="2-col2"
              >
                2-col2
              </p>
            </div>
          </td>
          <td
            style="max-width: 50px; width: 50px; min-width: 50px; box-shadow: inset 0 1px 0 rgba(0, 0, 0, 0.12);"
          >
            <div
              class="MuiStack-root css-1d9cypr-MuiStack-root"
            >
              <p
                class="MuiTypography-root MuiTypography-body2 css-foitbu-MuiTypography-root"
                title="2-col3"
              >
                2-col3
              </p>
            </div>
          </td>
        </tr>
        <tr
          style="height: 48px;"
        >
          <td
            style="max-width: 100%; width: 100%; min-width: 100%; box-shadow: inset 0 1px 0 rgba(0, 0, 0, 0.12);"
          >
            <div
              class="MuiStack-root css-1sg8hwy-MuiStack-root"
            >
              <p
                class="MuiTypography-root MuiTypography-body2 css-foitbu-MuiTypography-root"
                title="3-col1"
              >
                3-col1
              </p>
            </div>
          </td>
          <td
            style="max-width: 150px; width: 150px; min-width: 150px; box-shadow: inset 0 1px 0 rgba(0, 0, 0, 0.12);"
          >
            <div
              class="MuiStack-root css-1d9cypr-MuiStack-root"
            >
              <p
                class="MuiTypography-root MuiTypography-body2 css-foitbu-MuiTypography-root"
                title="3-col2"
              >
                3-col2
              </p>
            </div>
          </td>
          <td
            style="max-width: 50px; width: 50px; min-width: 50px; box-shadow: inset 0 1px 0 rgba(0, 0, 0, 0.12);"
          >
            <div
              class="MuiStack-root css-1d9cypr-MuiStack-root"
            >
              <p
                class="MuiTypography-root MuiTypography-body2 css-foitbu-MuiTypography-root"
                title=""
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                  data-testid="AccountTreeIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M22 11V3h-7v3H9V3H2v8h7V8h2v10h4v3h7v-8h-7v3h-2V8h2v3z"
                  />
                </svg>
              </p>
            </div>
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</DocumentFragment>
`;

exports[`Table renders correctly 2`] = `
<DocumentFragment>
  <div
    class="css-1qqflhf"
  >
    <table
      style="table-layout: fixed; width: 100%; text-align: left; border-collapse: collapse;"
    >
      <thead>
        <tr
          style="height: 48px;"
        >
          <th
            style="position: sticky; top: 0px; max-width: 100%; width: 100%; min-width: 100%; background-color: rgb(255, 255, 255); box-shadow: inset 0 -2px 0 rgba(0, 0, 0, 0.12); z-index: 1;"
          >
            <div
              class="MuiBox-root css-semc09"
            >
              <div
                class="MuiBox-root css-j6zvzo"
              >
                <span
                  class="MuiTypography-root MuiTypography-overline css-szitil-MuiTypography-root"
                  title="Column 1"
                >
                  Column 1
                </span>
              </div>
              <div
                class="MuiBox-root css-9qc0eg"
              >
                <button
                  class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeSmall css-1uncdf-MuiButtonBase-root-MuiIconButton-root"
                  tabindex="0"
                  type="button"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1vooibu-MuiSvgIcon-root"
                    data-testid="ArrowUpwardIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="m4 12 1.41 1.41L11 7.830h2V7.83l5.58 5.59L20 12l-8-8-8 8z"
                    />
                  </svg>
                  <span
                    class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                  >
                    <span
                      class="css-y4cjyz-MuiTouchRipple-ripple MuiTouchRipple-ripple MuiTouchRipple-rippleVisible"
                      style="width: 1px; height: 1px; top: -0.5px; left: -0.5px;"
                    >
                      <span
                        class="MuiTouchRipple-child MuiTouchRipple-childLeaving"
                      />
                    </span>
                  </span>
                </button>
              </div>
            </div>
          </th>
          <th
            style="position: sticky; top: 0px; max-width: 150px; width: 150px; min-width: 150px; background-color: rgb(255, 255, 255); box-shadow: inset 0 -2px 0 rgba(0, 0, 0, 0.12); z-index: 1;"
          >
            <div
              class="MuiBox-root css-semc09"
            >
              <div
                class="MuiBox-root css-j6zvzo"
              >
                <span
                  class="MuiTypography-root MuiTypography-overline css-szitil-MuiTypography-root"
                  title="Column 2"
                >
                  Column 2
                </span>
              </div>
              <div
                class="MuiBox-root css-9qc0eg"
              >
                <button
                  class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeSmall css-1pe4mpk-MuiButtonBase-root-MuiIconButton-root"
                  tabindex="0"
                  type="button"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1vooibu-MuiSvgIcon-root"
                    data-testid="SwapVertIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M16 17.01V10h-2v7.01h-3L15 21l4-3.99h-3zM9 3 5 6.99h3V14h2V6.99h3L9 3z"
                    />
                  </svg>
                  <span
                    class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                  />
                </button>
              </div>
            </div>
          </th>
          <th
            style="position: sticky; top: 0px; max-width: 50px; width: 50px; min-width: 50px; background-color: rgb(255, 255, 255); box-shadow: inset 0 -2px 0 rgba(0, 0, 0, 0.12); z-index: 1;"
          >
            <div
              class="MuiBox-root css-1jjjjm8"
            >
              <div
                class="MuiBox-root css-j6zvzo"
              >
                <span
                  class="MuiTypography-root MuiTypography-overline css-szitil-MuiTypography-root"
                  title="[object Object]"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                    data-testid="AccountTreeIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M22 11V3h-7v3H9V3H2v8h7V8h2v10h4v3h7v-8h-7v3h-2V8h2v3z"
                    />
                  </svg>
                </span>
              </div>
            </div>
          </th>
        </tr>
      </thead>
      <tbody>
        <tr
          style="height: 48px;"
        >
          <td
            style="max-width: 100%; width: 100%; min-width: 100%;"
          >
            <div
              class="MuiStack-root css-1sg8hwy-MuiStack-root"
            >
              <button
                class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeSmall css-1pe4mpk-MuiButtonBase-root-MuiIconButton-root"
                tabindex="0"
                type="button"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1vooibu-MuiSvgIcon-root"
                  data-testid="KeyboardArrowRightIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M8.59 16.59 13.17 12 8.59 7.41 10 6l6 6-6 6-1.41-1.41z"
                  />
                </svg>
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </button>
              <button
                class="MuiTypography-root MuiTypography-body2 MuiLink-root MuiLink-underlineNone MuiLink-button css-14wmvzk-MuiTypography-root-MuiLink-root"
                title="1-col1"
              >
                1-col1
              </button>
            </div>
          </td>
          <td
            style="max-width: 150px; width: 150px; min-width: 150px;"
          >
            <div
              class="MuiStack-root css-1d9cypr-MuiStack-root"
            >
              <p
                class="MuiTypography-root MuiTypography-body2 css-foitbu-MuiTypography-root"
                title="1-col2"
              >
                1-col2
              </p>
            </div>
          </td>
          <td
            style="max-width: 50px; width: 50px; min-width: 50px;"
          >
            <div
              class="MuiStack-root css-1d9cypr-MuiStack-root"
            >
              <p
                class="MuiTypography-root MuiTypography-body2 css-foitbu-MuiTypography-root"
                title=""
              />
            </div>
          </td>
        </tr>
        <tr
          style="display: none; height: 48px;"
        >
          <td
            style="max-width: 100%; width: 100%; min-width: 100%; box-shadow: inset 0 1px 0 rgba(0, 0, 0, 0.12);"
          >
            <div
              class="MuiStack-root css-kv8qlv-MuiStack-root"
            >
              <button
                class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeSmall css-1pe4mpk-MuiButtonBase-root-MuiIconButton-root"
                tabindex="0"
                type="button"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1vooibu-MuiSvgIcon-root"
                  data-testid="CheckBoxIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M19 3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.11 0 2-.9 2-2V5c0-1.1-.89-2-2-2zm-9 14-5-5 1.41-1.41L10 14.17l7.59-7.59L19 8l-9 9z"
                  />
                </svg>
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </button>
              <p
                class="MuiTypography-root MuiTypography-body2 css-foitbu-MuiTypography-root"
                title="2-col1"
              >
                2-col1
              </p>
            </div>
          </td>
          <td
            style="max-width: 150px; width: 150px; min-width: 150px; box-shadow: inset 0 1px 0 rgba(0, 0, 0, 0.12);"
          >
            <div
              class="MuiStack-root css-1d9cypr-MuiStack-root"
            >
              <p
                class="MuiTypography-root MuiTypography-body2 css-foitbu-MuiTypography-root"
                title="2-col2"
              >
                2-col2
              </p>
            </div>
          </td>
          <td
            style="max-width: 50px; width: 50px; min-width: 50px; box-shadow: inset 0 1px 0 rgba(0, 0, 0, 0.12);"
          >
            <div
              class="MuiStack-root css-1d9cypr-MuiStack-root"
            >
              <p
                class="MuiTypography-root MuiTypography-body2 css-foitbu-MuiTypography-root"
                title="2-col3"
              >
                2-col3
              </p>
            </div>
          </td>
        </tr>
        <tr
          style="height: 48px;"
        >
          <td
            style="max-width: 100%; width: 100%; min-width: 100%; box-shadow: inset 0 1px 0 rgba(0, 0, 0, 0.12);"
          >
            <div
              class="MuiStack-root css-1sg8hwy-MuiStack-root"
            >
              <p
                class="MuiTypography-root MuiTypography-body2 css-foitbu-MuiTypography-root"
                title="3-col1"
              >
                3-col1
              </p>
            </div>
          </td>
          <td
            style="max-width: 150px; width: 150px; min-width: 150px; box-shadow: inset 0 1px 0 rgba(0, 0, 0, 0.12);"
          >
            <div
              class="MuiStack-root css-1d9cypr-MuiStack-root"
            >
              <p
                class="MuiTypography-root MuiTypography-body2 css-foitbu-MuiTypography-root"
                title="3-col2"
              >
                3-col2
              </p>
            </div>
          </td>
          <td
            style="max-width: 50px; width: 50px; min-width: 50px; box-shadow: inset 0 1px 0 rgba(0, 0, 0, 0.12);"
          >
            <div
              class="MuiStack-root css-1d9cypr-MuiStack-root"
            >
              <p
                class="MuiTypography-root MuiTypography-body2 css-foitbu-MuiTypography-root"
                title=""
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                  data-testid="AccountTreeIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M22 11V3h-7v3H9V3H2v8h7V8h2v10h4v3h7v-8h-7v3h-2V8h2v3z"
                  />
                </svg>
              </p>
            </div>
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</DocumentFragment>
`;
